core/lib/types/src/commitment.rs

```diff
@@ -577,181 +573,134 @@ impl L1BatchCommitment {
+        }
     }
+}
 
-    #[derive(Debug, Clone, Serialize, Deserialize)]
-    struct InitialStorageTest {
-        pub key: String,
-        pub value: H256,
-    }
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+pub struct AuxCommitments {
+    pub events_queue_commitment: H256,
+    pub bootloader_initial_content_commitment: H256,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct CommitmentCommonInput {
+    pub l2_to_l1_logs: Vec<UserL2ToL1Log>,
+    pub rollup_last_leaf_index: u64,
+    pub rollup_root_hash: H256,
+    pub bootloader_code_hash: H256,
+    pub default_aa_code_hash: H256,
+    pub protocol_version: ProtocolVersionId,
+}
 
-    #[derive(Debug, Clone, Serialize, Deserialize)]
-    struct BlockAuxiliaryInput {
-        l2_l1_logs: Vec<L2ToL1Log>,
-        initial_writes: Vec<InitialStorageTest>,
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub enum CommitmentInput {
+    PreBoojum {
+        common: CommitmentCommonInput,
+        initial_writes: Vec<InitialStorageWrite>,
         repeated_writes: Vec<RepeatedStorageWrite>,
+    },
+    PostBoojum {
+        common: CommitmentCommonInput,
+        system_logs: Vec<SystemL2ToL1Log>,
+        state_diffs: Vec<StateDiffRecord>,
+        aux_commitments: AuxCommitments,
+    },
+}
+
+impl CommitmentInput {
+    pub fn common(&self) -> &CommitmentCommonInput {
+        match self {
+            Self::PreBoojum { common, .. } => common,
+            Self::PostBoojum { common, .. } => common,
+        }
     }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct L1BatchCommitmentArtifacts {
+    pub commitment_hash: L1BatchCommitmentHash,
+    pub l2_l1_merkle_root: H256,
+    pub compressed_state_diffs: Option<Vec<u8>>,
+    pub compressed_initial_writes: Option<Vec<u8>>,
+    pub compressed_repeated_writes: Option<Vec<u8>>,
+    pub zkporter_is_available: bool,
+    pub aux_commitments: Option<AuxCommitments>,
+}
+
+#[cfg(test)]
+mod tests {
+    use serde::{Deserialize, Serialize};
+
+    use super::{
+        CommitmentInput, L1BatchAuxiliaryOutput, L1BatchCommitment, L1BatchCommitmentHash,
+        L1BatchMetaParameters, L1BatchPassThroughData,
+    };
 
     #[derive(Debug, Serialize, Deserialize)]
     struct CommitmentTest {
+        input: CommitmentInput,
         pass_through_data: L1BatchPassThroughData,
-        auxiliary_input: BlockAuxiliaryInput,
         meta_parameters: L1BatchMetaParameters,
-        expected_outputs: ExpectedOutput,
+        auxiliary_output: L1BatchAuxiliaryOutput,
+        hashes: L1BatchCommitmentHash,
     }
 
-    #[test]
-    fn commitment_test() {
+    fn run_test(test_name: &str) {
         let zksync_home = std::env::var("ZKSYNC_HOME").unwrap_or_else(|_| ".".into());
         let path = std::path::Path::new(&zksync_home)
-            .join("etc/commitment_tests/zksync_testharness_test.json");
+            .join(format!("etc/commitment_tests/{test_name}.json"));
```

Comment:
Sort of off-topic: Specifying paths outside of the package makes it really difficult to publish it because package tests are no longer even theoretically self-contained. (I believe we have other places with similar issues, but the less such places, the better.) In this case, it would be easy to move (or symlink, if they are used anywhere else) the necessary files inside the package directory.